libsplinter/src/node_registry/yaml/local.rs

```diff
@@ -135,7 +136,17 @@ impl LocalYamlNodeRegistry {
                 Box::new(err),
             )
         })?;
-        std::fs::write(&file_backend.file_path, &output).map_err(|err| {
+
+        let mut file = File::create(&file_backend.file_path).map_err(|err| {
+            NodeRegistryError::general_error_with_source(
+                &format!(
+                    "Failed to open YAML registry file '{}'",
+                    file_backend.file_path
+                ),
+                Box::new(err),
+            )
+        })?;
+        file.write_all(&output).map_err(|err| {
```

Comment:
Since you're cleaning this up a bit, you might use `serde_yaml::to_writer` to write to the file you created, instead of creating the bytes.

Reply:
Unfortunately, `serde_yaml::to_writer` consumes the `Write` and `File` isn't easily cloneable, so that isn't a clean replacement.